src/main/scala/devices/debug/Periphery.scala

```diff
@@ -16,24 +16,39 @@ import freechips.rocketchip.jtag._
 import freechips.rocketchip.util._
 import freechips.rocketchip.tilelink._
 
+/** Protocols used for communicating with external debugging tools */
+sealed trait DebugExportProtocol
+case object DMI extends DebugExportProtocol
+case object JTAG extends DebugExportProtocol
+case object CJTAG extends DebugExportProtocol
+case object APB extends DebugExportProtocol
+
 /** Options for possible debug interfaces */
-case object ExportDebugDMI extends Field[Boolean](true)
-case object ExportDebugJTAG extends Field[Boolean](false)
-case object ExportDebugCJTAG extends Field[Boolean](false)
-case object ExportDebugAPB extends Field[Boolean](false)
-case object ExportDisableDebug extends Field[Boolean](false)
+case class DebugAttachParams(
+  protocol: DebugExportProtocol = DMI,
+  externalDisable: Boolean = false,
+  masterWhere: BaseSubsystemBusAttachment = FBUS,
+  slaveWhere: BaseSubsystemBusAttachment = CBUS
+) {
+  def dmi   = protocol == DMI
```

Comment:
in the future we may want to allow more than one protocol. For example JTAG AND APB. Does this refactor consider this?

Reply:
Could we make the protocol argument to DebugAttachParams be a `Seq[DebugExportProtocol]`?

Reply:
I just had an offline discussion with @hcook about this issue and this is exactly what he proposed.

Reply:
I made them a `Set`